### PR TITLE
Add a flag to skip "rake seed:all" during "rake build"

### PR DESCRIPTION
**Pull request by davidsbailey**
In a recent run on my local machine it takes 5m55s to `rake bulid` including `rake seed:all`, and only 2m30s to `rake build` without `rake seed:all` (with `skip_seed_all` set).

Reviewers, please feel free to improve my warning in locals.yml.default about what this flag does.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Is the cost-gain benefit from skipping seeding in pegasus also positive (by default)? If so, consider doing the same for it.

File: lib/rake/build.rake [hunk 1]
```diff
@@ -108,8 +108,13 @@ namespace :build do
           end
         end
 
-        HipChat.log 'Seeding <b>dashboard</b>...'
-        RakeUtils.rake 'seed:all'
+        # Allow developers to skip the time-consuming step of seeding the dashboard DB.
+        if (rack_env?(:development) && CDO.skip_seed_all)
+          HipChat.log 'Not seeding <b>dashboard</b> due to CDO.skip_seed_all...'
+        else
+          HipChat.log 'Seeding <b>dashboard</b>...'
```
Comment (pcardune) [inline lib/rake/build.rake:118]: Maybe you can add a message here about modifying `locals.yml` if this is taking too long.

**Comment by pcardune** [comment]
I like it, but I don't know enough about the implications of this change to do a proper review.

**Comment by ashercodeorg** [comment]
Looking forward to this, it'll save me loads of time. It does mean that knowledge of what `rake seed:all` does should be better known across the team (is it widespread?).

**Comment by philbogle** [comment]
This is a helpful option as long as developers remember to manually run seed:all when they make a change to one of its inputs.

In the future, we might want to consider a make-based option that we could leave on all of the time.  Given an explicitly list of inputs and generated files, the make system could ensure that the outputs are regenerated if and only if the inputs are newer.  That would eliminate the possibility of human error.

**Comment by davidsbailey** [comment]
@ashercodeorg: pegasus's `rake seed:migrate` takes 3 seconds for me -- probably not worth skipping. dashboard's `rake seed:all` is the obvious low-hanging fruit. To address your other concern, I've added warnings explaining what exactly getting skipped every time `rake build` skips it, and also in locals.yml.default.

**Pull request merged**